src/market/client.rs

```diff
@@ -0,0 +1,93 @@
+use {
+    super::{
+        error::ApiError,
+        models::{
+            item::{Item, ItemsPayload},
+            item_info::{ItemInfo, ItemInfoPayload},
+            orders::{Order, OrderPayload},
+            statistic_item::{StatisticItem, StatisticItemPayload},
+        },
+    },
+    std::fmt::Display,
+};
+
+#[derive(Default, Debug, Clone)]
+pub struct Client {
+    session: reqwest::Client,
+}
+
+impl Client {
+    pub fn new() -> Self {
+        Default::default()
+    }
+}
+
+impl Client {
+    /// Fetches the statistics of an item via its url_name
+    pub async fn item_statistics(&self, item_url: &str) -> Result<StatisticItem, ApiError> {
+        let response = self
+            .session
+            .get(format!(
+                "https://api.warframe.market/v1/items/{item_url}/statistics"
+            ))
+            .send()
+            .await?;
+
+        if response.status().is_success() {
+            let json_result = response.json::<StatisticItemPayload>().await?;
+            Ok(json_result.payload)
+        } else {
+            Err(response.status().into())
+        }
+    }
+
+    /// Fetches info about an item via its url_name
+    pub async fn item_info(&self, item_url: &str) -> Result<ItemInfo, ApiError> {
+        let response = self
+            .session
+            .get(format!("https://api.warframe.market/v1/items/{item_url}"))
+            .send()
+            .await?;
+
+        if response.status().is_success() {
+            let json_result = response.json::<ItemInfoPayload>().await?;
+            Ok(json_result.payload.item)
+        } else {
+            Err(response.status().into())
+        }
+    }
+
+    /// Fetches all tradable items
+    pub async fn items(&self) -> Result<Vec<Item>, ApiError> {
```

Comment:
you'll probably wanna ratelimit this one yourself or catch ratelimit errors (HTTP 429), and cache this endpoint, cause it tends to be slow if you overuse it

Reply:
I see. Yea, Ratelimiting is prolly a good idea. Alltho, I'd make the cache optional via a feature flag - maybe someone wants to implement their own?
Yeaaa, I kinda messed up the first rebase. Will do as soonnas I'm home

Reply:
well, problem on the cache is i'm more getting at you probably wanna cache it so if you get a 429, you can return the last result and add some additional response info that you returned from cache due to ratelimit

Reply:
i can rebase for you if you want

Reply:
i did some rebase